### ISSUE TITLE
Selecting macOS SDK in some cases

### ISSUE THREAD

**Issue by jakirkham** [conda-forge/conda-smithy#769]
Apparently Qt will require the macOS 10.11 SDK. So we will need to figure out a way to allow this.

cc @mingwandroid

**Comment by isuruf** [comment]
This should be moved to `conda-forge-ci-setup-feedstock`. We can already override the variables in conda_build_config.yaml.

**Comment by jakirkham** [comment]
Hmm...how would you propose setting the macOS SDK version to use in a feedstock?

**Comment by isuruf** [comment]
add it to `recipe/conda_build_config.yaml`

**Comment by jakirkham** [comment]
How would `conda-forge-ci-setup` pick up on that change?

**Comment by isuruf** [comment]
After rendering it should be in `${CONFIG}.yaml` as well. We could read the yaml file and get the value.

Change the line https://github.com/conda-forge/conda-forge-ci-setup-feedstock/blob/master/recipe/run_conda_forge_build_setup_osx#L13
```
export MACOSX_DEPLOYMENT_TARGET=`python -c "import yaml;print(yaml.load(open('$CONFIG.yaml'))['MACOSX_DEPLOYMENT_TARGET'][0])"`
```

**Comment by jakirkham** [comment]
So right now we set the Travis CI image based on what macOS SDK we want. The current version `xcode6.4` has macOS SDKs for 10.9 and 10.10. Presumably we would want `xcode7.3` or `xcode8` for macOS 10.11's SDK.

ref: https://docs.travis-ci.com/user/reference/osx/

**Comment by isuruf** [comment]
True, but if we use the new anaconda compilers it doesn't matter which image we use right?

**Comment by jakirkham** [comment]
It matters in term of having the macOS SDK. Anaconda is not supplying those with the compilers.

**Comment by mingwandroid** [comment]
If you were to make conda packages from the SDKs then you'd be able to add them as build deps (or just install them all into the root env?). I'm not sure what's best here.

**Comment by isuruf** [comment]
> It matters in term of having the macOS SDK. Anaconda is not supplying those with the compilers.

Right. This SDK should be downloaded by `conda-forge-ci-setup` right?

**Comment by jakirkham** [comment]
So we could avoid the download, which seems preferable, if we choose the correct Travis CI image. It's really just a matter of looking at the `recipe/conda_build_config.yaml` for the macOS version and then following a predefined mapping to figure out what image to inject into the `.travis.yml` file.

**Comment by jakirkham** [comment]
> If you were to make conda packages from the SDKs then you'd be able to add them as build deps (or just install them all into the root env?).

Right, PR ( https://github.com/conda-forge/staged-recipes/pull/5382 ) would help with that.
